src/libaktualizr/uptane/imagerepository.cc

```diff
@@ -239,6 +250,8 @@ void ImageRepository::updateMeta(INvStorage& storage, const IMetadataFetcher& fe
     }
 
     checkTimestampExpired();
+    timestamp_updated_snapshot_version = last_snapshot_version_in_timestamp != timestamp.snapshot_version();
```

Comment:
@detsch I am just wondering if setting `timestamp_updated_snapshot_version = true` just after #249 would work, no need the static variable definition at #219.

Reply:
> @detsch I am just wondering if setting `timestamp_updated_snapshot_version = true` just after 249 would work, no need the static variable definition at 219.

It would simplify the implementation, but `timestamp_updated_snapshot_version` would be true for any change in the timestamp (e.g., updating expiration), even when the snapshot version was not updated as well.

Reply:
We have to support the use-case when the new/received timestamp has the same version as the latest local one, but their content is different. That's why we have this condition `timestamp_stored_signature != timestamp.signature()` in addition to `local_version < remote_version`.
In this case, we update the locally stored  timestamp (#249) and we have to check the remote snapshot anyway even if its version is the same as the latest local because its content may point to the different targets.
This is the case when a device is switched from one tag to another. Each tag has its own set of TUF metadata with the same version but different content in timestamp/snapshot/targets.

Your right, when timestamp is just refreshed (no tag switching) then aklite will fetch the same snapshot again. However, it's not critical and if it's the same snapshot as the latest local one then targets won't be fetched and a size of snapshot is not so big.

Reply:
Actually, I am wrong even snapshot won't be pulled since snapshot hash is the same.

```
 try {
        verifySnapshot(image_snapshot_stored, true);
        fetch_snapshot = false;
        LOG_DEBUG << "Skipping Image repo Snapshot download; stored version is still current.";
```

Reply:
The updated/refreshed timestamp has a snapshot hash that matches hash of the locally stored snapshot.

Reply:
Since we may have a same snapshot version with different contents, something like `timestamp_updated_snapshot_hashes = last_snapshot_hashes_in_timestamp != timestamp.snapshot_hashes();` could be more appropriate. I.e., look at the hashes directly, instead of comparing versions.

Reply:
Could be, or the hash checking functionality inside  `verifySnapshot` can be reused - return a boolean indicating whether hash is the same (`hash_changed`).

Then, if timestamp_updated and hash_changed then log_debug else log_error.

I just don't like that static variable. Maybe as an alternative the `last_snapshot_hashes_in_timestamp` could be a class member updated in `verifySnapshot`.

The problem is that the `hashes` is a vector, though.

Reply:
OK, I was missing something, the solution is simple and we don't need an static variable nor a class member. We can use the previously loaded timestamp, that is available at the very beginning of the method call, before being reset inside the `updateRoot(storage, fetcher)` call. I'm adjusting the code accordingly.
BTW, there seems to be an issue here: `const auto timestamp_stored_signature{timestamp.isInitialized() ? timestamp.signature() : ""};`. The timestamp is never initialized at that point (because of the `updateRoot` call), and we end up always storing the timestamp metadata file, even if there are no changes. I'll take a better look to confirm this, and prepare a patch for this issue as well.

Reply:
Created #67

Reply:
That's indeed the issue. Thanks for fixing it.